lua/bufferline/render.lua

```diff
@@ -137,8 +137,8 @@ local function render()
         iconPrefix = ''
         icon = number_text .. (#number_text > 1 and '' or ' ')
       else
-        local iconChar, iconHl = get_icon(buffer_name, vim.fn.getbufvar(buffer_number, '&filetype'))
-        iconPrefix = status == 'Inactive' and hl('BufferInactive') or hl(iconHl or ('Buffer' .. status))
+        local iconChar, iconHl = get_icon(buffer_name, vim.fn.getbufvar(buffer_number, '&filetype'), status)
```

Comment:
Now, because we can determine the background color of the icon guaranteed, I have simplified the highlight statement back to a similar version of what it was.